src/OpenConext/Profile/Value/Organization.php

```
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\Profile\Value;
+
+use Assert\AssertionFailedException;
+use OpenConext\Profile\Assert;
+
+final class Organization
+{
+    /**
+     * @var string[]
+     */
+    private $displayName;
+
+    /**
+     * @var string[]
+     */
+    private $name;
+
+    /**
+     * @var Logo
+     */
+    private $logo;
+
+    public function __construct(array $displayName, array $name, Logo $logo)
+    {
+        Assert::allString(array_keys($displayName), 'DisplayName translations must be indexed by locale');
+        Assert::allNotBlank(array_keys($displayName), 'Locales may not be blank');
+        Assert::allString(array_keys($name), 'Name translations must be indexed by locale');
+        Assert::allNotBlank(array_keys($name), 'Locales may not be blank');
+
+        $this->displayName = $displayName;
+        $this->name = $name;
+        $this->logo = $logo;
+    }
+
+    public static function fromArray(array $json): Organization
+    {
+        Assert::keysArePresent($json, ['display_name', 'name', 'logo']);
+        return new self($json['display_name'], $json['name'], Logo::fromArray($json['logo']));
+    }
+
+    public function getDisplayName(string $locale): string
+    {
+        $displayNameLocale = $this->displayName[$locale];
+        if (!empty($displayNameLocale)) {
+            return $displayNameLocale;
+        }
+
+        $localeName = $this->name[$locale];
+        if (!empty($localeName)) {
+            return $localeName;
+        }
+
+        $englishName = $this->name['en'];
+        if (!empty($englishName)) {
+            return $englishName;
+        }
+
+        return '';
+    }
+
+    /**
+     * @return false|Logo
+     */
+    public function getLogo()
+    {
+        if (!$this->hasLogo()) {
+            return false;
+        }
```

Having a mixed return type is not encouraged. I'd simply let the method throw an exception when the logo is not set. Which should never be the case, as you can test this using the 'hasLogo' method.

---

seems so obvious in hindsight, good call 💯